metrics/ci_integrations/test/source/github_actions/config/validator/github_actions_source_validator_test.dart

```diff
@@ -0,0 +1,556 @@
+
+    final config = GithubActionsSourceConfig(
+      accessToken: accessToken,
+      repositoryOwner: repositoryOwner,
+      repositoryName: repositoryName,
+      workflowIdentifier: workflowId,
+      jobName: jobName,
+      coverageArtifactName: coverageArtifactName,
+    );
+    final auth = BearerAuthorization(accessToken);
+    final validationDelegate = _GithubActionsSourceValidationDelegateMock();
+    final validationResultBuilder = _ValidationResultBuilderMock();
+    final validator = GithubActionsSourceValidator(
+      validationDelegate,
+      validationResultBuilder,
+    );
+    final field = GithubActionsSourceConfigField.workflowIdentifier;
+    final validationResult = ValidationResult({
+      field: result,
+    });
+
+    PostExpectation<Future<InteractionResult<GithubToken>>> whenValidateAuth() {
+      return when(validationDelegate.validateAuth(auth));
+    }
+
+    PostExpectation<Future<InteractionResult<void>>>
+        whenValidateRepositoryOwner() {
+      whenValidateAuth().thenSuccessWith(githubToken, message);
+
+      return when(
+        validationDelegate.validateRepositoryOwner(repositoryOwner),
+      );
+    }
+
+    PostExpectation<Future<InteractionResult<void>>>
+        whenValidateRepositoryName() {
+      whenValidateRepositoryOwner().thenSuccessWith(null, message);
+
+      return when(
+        validationDelegate.validateRepositoryName(
+          repositoryName: repositoryName,
+          repositoryOwner: repositoryOwner,
+        ),
+      );
+    }
+
+    PostExpectation<Future<InteractionResult<void>>> whenValidateWorkflowId() {
+      whenValidateRepositoryName().thenSuccessWith(null, message);
+
+      return when(
+        validationDelegate.validateWorkflowId(workflowId),
+      );
+    }
+
+    tearDown(() {
+      reset(validationDelegate);
+      reset(validationResultBuilder);
+    });
+
+    test(
+      "throws an ArgumentError if the given validation delegate is null",
+      () {
+        expect(
+          () => GithubActionsSourceValidator(null, validationResultBuilder),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      "throws an ArgumentError if the given validation result builder is null",
+      () {
+        expect(
+          () => GithubActionsSourceValidator(validationDelegate, null),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      "creates a new instance with the given parameters",
+      () {
+        final validator = GithubActionsSourceValidator(
+          validationDelegate,
+          validationResultBuilder,
+        );
+
+        expect(validator.validationDelegate, equals(validationDelegate));
+        expect(
+          validator.validationResultBuilder,
+          equals(validationResultBuilder),
+        );
+      },
+    );
+
+    test(
+      ".validate() sets empty results with the unknown field validation result with the 'token invalid' additional context if the access token is null",
```

Comment:
Should we test that this method returns a validation result if the access token is `null`?